jcl/src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/TypeLayoutCheckHelper.java

```diff
@@ -0,0 +1,234 @@
+/*[INCLUDE-IF JAVA_SPEC_VERSION <= 17]*/
+/*******************************************************************************
+ * Copyright (c) 2021, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package jdk.internal.foreign.abi;
+
+import java.util.List;
+import java.lang.invoke.MethodType;
+
+import jdk.incubator.foreign.CLinker.TypeKind;
```

Comment:
TypeKind has been removed in JDK18

Reply:
ignore

Reply:
This file is excluded since Java 18 given the method types are directly inferred from the function descriptor in which case the check on  is taken over on OpenJDK side.